scrapingundsoup_ex.ipynb

```diff
@@ -0,0 +1,176 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 330,
+   "metadata": {
+    "autoscroll": false,
+    "ein.hycell": false,
+    "ein.tags": "worksheet-0",
+    "slideshow": {
+     "slide_type": "-"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Our jupyter/datascience-notebook Docker container comes with \n",
+    "# BeautifulSoup4 and requests, both popular libraries!\n",
+    "\n",
+    "from bs4 import BeautifulSoup\n",
+    "import requests"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 331,
+   "metadata": {
+    "autoscroll": false,
+    "ein.hycell": false,
+    "ein.tags": "worksheet-0",
+    "slideshow": {
+     "slide_type": "-"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "START_URL = 'https://brickset.com/sets/year-2016'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 332,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_titles(soup):    \n",
+    "    \"\"\" Returns a list of titles on the page \"\"\"\n",
+    "    # the \"soup\" parameter is of the type that is\n",
+    "    # returned by Beautiful Soup when it parses HTML.\n",
+    "    # The function should then use the object to\n",
+    "    # extract a list of titles (of the lego sets)\n",
+    "    #\n",
+    "    # Lookup the documentation for Beautiful Soup\n",
+    "    # Figure out how to select the text of the title\n",
+    "    # of each legoset. A title should look like: \n",
+    "    # \"10252: Volkswagen Beetle\"\n",
+    "    dict_title = {}\n",
+    "    for i in range(0,51):\n",
```

Comment:
An easier way to get to the list: 

```{py}
list_titles = [h.get_text() for h in soup.find_all("h1")]
```

But, in generally, your selection is not ideal, because you're getting ALL h1's, but you don't want ALL of them! Therefore, use more specific CSS selectors (via soup.select) to only get the ones you want, then you don't have to try and filter later (which is fragile!).